app/models/concerns/client_search.rb

```diff
@@ -26,6 +26,12 @@ def self.text_searcher(text, sorted:, resolve_for_join_query: false)
       max_pk = 2_147_483_648 # PK is a 4 byte signed INT (2 ** ((4 * 8) - 1))
       term_is_possibly_pk = numeric ? text.to_i < max_pk : false
 
+      # If alphanumeric search term matches a Scan Card code, return immediately
+      if alpha_numeric && HmisEnforcement.hmis_enabled?
+        matching_scan_card = Hmis::ScanCardCode.find_by(value: text)
+        return where(sa[:id].eq(matching_scan_card.client_id)) if matching_scan_card
+      end
```

Comment:
If a scan card can be 'S' + hex chars, it could be the same as someone's name right? For example "Sade" is a human name and a hex card code. These collisions are a tiny edge case but could cause very unexpected search results.

Reply:
Yes it is possible though very unlikely. We could throw another random number on there to reduce the likelihood, since names generally should not have numbers in there. sound good ?